File: 165.java
class Solution {
    public int compareVersion(String version1, String version2) {
        int sum1 = 0, sum2 = 0;
        int n1 = version1.length(), n2 = version2.length();
        int i = 0, j = 0;
        while (i < n1 || j < n2) {
            sum1 = 0;
            sum2 = 0;
            while (i < n1 && version1.charAt(i) != '.') {
                sum1 = sum1 * 10 + version1.charAt(i) - '0';
                i++;
            }
            while (j < n2 && version2.charAt(j) != '.') {
                sum2 = sum2 * 10 + version2.charAt(j) - '0';
                j++;
            }
            if (sum1 > sum2) {
                return 1;
            }
            if (sum1 < sum2) {
                return -1;
            }
            i++;
            j++;
        }
        return 0;
    }
}

File: Python/165.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        version1_l = version1.split(".")
        version2_l = version2.split(".")

        index = 0
        while index < len(version1_l) or index < len(version2_l):
            cur_v1 = int(version1_l[index]) if index < len(version1_l) else 0
            cur_v2 = int(version2_l[index]) if index < len(version2_l) else 0
            if cur_v1 < cur_v2:
                return -1
            elif cur_v1 > cur_v2:
                return 1
            index += 1
        return 0
